src/Uno.UI/UI/Xaml/Controls/BitmapIcon.cs

```diff
@@ -1,36 +1,48 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Text;
+using Windows.Foundation;
 using Windows.UI.Xaml.Data;
 using Windows.UI.Xaml.Media;
 
 namespace Windows.UI.Xaml.Controls
 {
 	public partial class BitmapIcon : IconElement
 	{
-		private readonly Image _image;
-		private readonly Grid _grid;
+		private Image _image;
+		private Grid _grid;
 
 		public BitmapIcon()
 		{
-			Foreground = SolidColorBrushHelper.Black;
+			this.SetValue(ForegroundProperty, SolidColorBrushHelper.Black, DependencyPropertyValuePrecedences.Inheritance);
 
-			_image = new Image {
-				Stretch = Media.Stretch.Uniform,
+		}
+
+		protected override Size MeasureOverride(Size availableSize)
+		{
+			if (_image == null)
+			{
+				_image = new Image
+				{
+					Stretch = Media.Stretch.Uniform,
 #if !NET461
-				MonochromeColor = (Foreground as SolidColorBrush)?.Color
+					MonochromeColor = (Foreground as SolidColorBrush)?.Color
```

Comment:
should `ShowAsMonochrome` be considered here?

Reply:
Indeed it should!